src/07-reference.md

```diff
@@ -0,0 +1,580 @@
+# Language reference
+
+This is a living specification of the current set of built-ins in the `.lurk` package.
+
+## Built-in atoms
+
+### `nil`
+
+`nil` is boolean false ans also the empty list.
+
+It has its own type differently from other symbols.
+
+```
+user> (eq nil '())
+[3 iterations] => t
+user> (cdr '(1))
+[2 iterations] => nil
+user> (if nil "true" "false")
+[3 iterations] => "false"
+```
+
+### `t`
+
+`t` is boolean true. In boolean contexts like `if`, anything that isn't `nil` is considered false, but `t` is generally used.
+
+```
+user> (eq 1 1)
+[2 iterations] => t
+user> (if t "true" "false")
+[3 iterations] => "true"
+```
+
+## Built-in functions
+
+### `atom`
+
+`(atom x)` returns `nil` if `x` is a pair, and `t` otherwise.
+
+```
+user> (atom nil)
+[2 iterations] => t
+user> (atom t)
+[2 iterations] => t
+user> (atom (cons 1 2))
+[4 iterations] => nil
+user> (atom '(1 2 3))
+[2 iterations] => nil
+```
+
+### `begin`
+
+`(begin forms...)` evaluates each of its arguments in sequence and return the last evaluation.
+
+```
+user> (begin 1 2 3)
+[6 iterations] => 3
+user> ((lambda (x) (begin (emit x) (+ x x))) 1)
+1
+[8 iterations] => 2
+```
+
+### `bignum`
+
+`(bignum x)` tries to convert `x` to a big num. Can only be used with commitments currently. Returns `<Err CantCastToBigNum>` if the types are incompatible.
+
+```
+user> (bignum (commit 1))
+[3 iterations] => #0x35ff7b6fb1e777a95aadced4e080fa9c024a47d7b1f81d217fa04a2a494fbb
+```
+
+### `car`
+
+`(car cons-cell)` returns the first element of its argument if it is a pair. Also works with lists and strings, returning its first element. Returns `<Err NotCons>` otherwise.
+
+```
+user> (car (cons 1 2))
+[4 iterations] => 1
+user> (car '(1 2 3))
+[2 iterations] => 1
+user> (car (strcons 'a' "b"))
+[4 iterations] => 'a'
+user> (car "abc")
+[2 iterations] => 'a'
+```
+
+### `cdr`
+
+`(cdr cons-cell)` returns the second element of its argument if it is a pair. Also works with lists and strings, returning its remainder. Returns `<Err NotCons>` otherwise.
```

Comment:
```suggestion
`(cdr cons-cell)` returns the second element of its argument if it is a pair. Also works with strings, returning its tail. Returns `<Err NotCons>` otherwise.
```